cli/src/semgrep/engine.py

```diff
@@ -63,7 +63,8 @@ def get_cpu_count() -> int:
     def default_jobs(self) -> int:
         if self == EngineType.PRO_INTERFILE:
             return 1
-        return self.get_cpu_count()
+        # Maxing out number of cores used to 16 if more not requested to not overload on large machines
+        return min(16, self.get_cpu_count())
```

Comment:
And we shouldn't be doing this silently, there should be warning being logged here IMO.